src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -1389,68 +1390,168 @@ int SketchObject::movePoint(int GeoId, PointPos PosId, const Base::Vector3d& toP
     return lastSolverStatus;
 }
 
+template <>
+Base::Vector3d SketchObject::getPointForGeometry<>(const Part::GeomPoint *geomPoint, PointPos PosId)
+{
+    if (PosId == PointPos::start || PosId == PointPos::mid || PosId == PointPos::end)

```

Comment:
Why for all PointPos but none you are returning the point, but for none you are returning Base::Vector3d()? How Base::Vector3d() is better than returning the point ?

Reply:
@PaddleStroke selecting a range when commenting makes it easier to read on the PR page as you then can see all relevant lines directly in the comment on the PR (not just the selected line and three above) See step (5) on [adding-comments-to-a-pull-request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request#adding-comments-to-a-pull-request)

For making it easier for the next person, here's a [link to the comment in the code](https://github.com/FreeCAD/FreeCAD/pull/15811/files/05448b980a35152eafe3990740a105b7fa556367#r1868916354).

Reply:
IIRC it is just consistent with what was being done earlier. That said most likely there should be an exception thrown (maybe with open/close principle applied).

Can you think of a case where `PointPos::None` could be useful with this method?

Reply:
It would just make the function more forgiving.
It's just that the function has a if to test if it's start end or mid. end or mid have not much more justification to be there than none. So I think we can just drop the if. If one querry the point geo the pos he gives should not matter.